tools/mascot/src/main/java/dk/alexandra/fresco/tools/mascot/field/AuthenticatedElement.java

```diff
@@ -0,0 +1,73 @@
+package dk.alexandra.fresco.tools.mascot.field;
+
+import dk.alexandra.fresco.tools.mascot.arithm.Addable;
+
+import java.math.BigInteger;
+
+public class AuthenticatedElement implements Addable<AuthenticatedElement> {
```

Comment:
The `Addable` interface exists largely because there are several batch addition methods in the `ArithmeticCollectionUtils`. Neither subtraction nor multiplication are used as batch operators anywhere so I did not make a general arithmetic interface and kept things minimal instead.